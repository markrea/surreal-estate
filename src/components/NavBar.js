import React from "react";
import { Link } from "react-router-dom";
import FacebookLogin from "react-facebook-login";

import "../styles/NavBar.css";

const NavBar = ({ onLogin, onLogout, userID}) => {
  return (
    <div className="navbar" data-testid="nav-bar-id">
      <img
        className="SE-logo"
        data-testid="logo-id"
        src="https://mcrcodes.s3.eu-west-2.amazonaws.com/course/surreal-estate/logo.png"
        alt="SELogo"
      />
      <ul className="navbar-links">
        <li>
          <Link to="/" className="navbar-links-item" data-testid="view-properties-id">
            View Properties
          </Link>
        </li>
        <li>
          <Link to="/add-property" className="navbar-links-item" data-testid="add-property-id">
            Add a Property
          </Link>
        </li>
        <li className="facebook-button">
        {!userID ? (
          <FacebookLogin
            appId="287306666028054"
            autoLoad={true}
            fields="name,email,picture"
            callback={onLogin}
          />
        ) : (
          <button onClick={onLogout}>Sign Out</button>
        )}
        </li>
      </ul>
      
    </div>
  );
};

export default NavBar;
