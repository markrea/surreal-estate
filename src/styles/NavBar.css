.navbar {
  grid-column: 1 / 4;
  grid-row: 1 / 2;
  display: flex;
  background-color: white;
  border-bottom: 1px solid lightgrey;
  justify-content: space-evenly;
}


.SE-logo {
  max-width: 35%;
  overflow: hidden;
  object-fit: contain;
}
.navbar-links {
  display: flex;
  align-items: center;
  padding: 0%;
  list-style-type: none;
  
 
  
}
.navbar-links-item {
  display: flex;
  flex-direction: row;
  flex-wrap: wrap;
  margin: auto;
  padding: 20px;
  text-decoration: none;
  color: black;
  font-size: 3vw;

}

.navbar-links-item:focus {
  text-decoration: underline;
}

@media screen and (min-width: 620px) {
  .navbar-links-item {
    font-size: 18px;
    padding: 0 8vw 0 30px;
    width: 150px;
  }
  .SE-logo {
    width: 247px;
    height: auto;
  }
}
.facebook-button {
  width: 80%;
}
