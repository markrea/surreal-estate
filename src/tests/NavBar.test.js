import React from "react";
import { render } from "@testing-library/react";
import NavBar from "../components/NavBar";

test("component renders correctly", () => {
  const { asFragment } = render(<NavBar />);
  const component = asFragment();
  expect(component).toMatchSnapshot();
});
