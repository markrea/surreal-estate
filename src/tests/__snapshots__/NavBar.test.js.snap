// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Navbar renders correctly 1`] = `
<DocumentFragment>
  <div
    class="navbar"
    data-testid="nav-bar-id"
  >
    <img
      alt="SELogo"
      class="SE-logo"
      data-testid="logo-id"
      src="https://mcrcodes.s3.eu-west-2.amazonaws.com/course/surreal-estate/logo.png"
    />
    <ul
      class="navbar-links"
    >
      <li>
        <a
          class="navbar-links-item"
          data-testid="view-properties-id"
          href="/"
        >
          View Properties
        </a>
      </li>
      <li>
        <a
          class="navbar-links-item"
          data-testid="add-property-id"
          href="/add-property"
        >
          Add a Property
        </a>
      </li>
      <li
        class="facebook-button"
      >
        <button>
          Sign Out
        </button>
      </li>
    </ul>
  </div>
</DocumentFragment>
`;
